internal/doctor/core.go

```diff
@@ -0,0 +1,142 @@
+package doctor
+
+import (
+	"github.com/gin-gonic/gin"
+	"hospitalApp/internal/patient"
+	"log"
+	"net/http"
+	"time"
+)
+
+type CreateRequestDoctorData struct {
+	DoctorId  uint   `json:"doctor_id"`
+	Name      string `json:"name"`
+	ContactNo string `json:"contact_no"`
+}
+
+type UpdateRequestDoctorData struct {
+	DoctorId  uint   `json:"doctor_id"`
+	Name      string `json:"name"`
+	ContactNo string `json:"contact_no"`
+}
+
+type UpdateRequestPatientData struct {
+	DoctorId uint            `json:"doctor_id"`
+	Patient  patient.Patient `json:"patient"`
+}
+
+func GetEntityByName(context *gin.Context) {
+	name := context.Query("name")
+	doctor, err := FindUserByName(name)
+	log.Printf("Doctor data get by Name(%q) : %+v", name, doctor)
+	if err != nil {
+		context.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	}
+
+	if doctor.DoctorId == 0 {
+		context.JSON(http.StatusOK, gin.H{"data": ""})
+		return
+	}
+	context.JSON(http.StatusOK, gin.H{"data": doctor})
+}
+
+func GetAllEntities(context *gin.Context) {
```

Comment:
You can define a struct for the core functions also

Reply:
Done